### ISSUE TITLE
Audio crackling on Linux client

### ISSUE THREAD

**Issue by selfdocumentingcode** [moonlight-stream/moonlight-qt#785]
**Describe the bug**
The audio crackles when streaming through the Linux client. 

**Steps to reproduce**
Stream anything through the Linux client. 

**Affected games**
Every one of the 5+ games I tried to so far. 

**Other Moonlight clients**
There is no audio crackling when using the Windows client. (same client pc and same host pc)

**Moonlight settings (please complete the following information)**
The only settings I changed are related to streaming resolution, fps and bitrate. 
The audio still crackles, maybe slight less, on for example 720p 30fps than 1440p 60fps. Tried various combinations of resolution, fps and default, low and high bitrate values. 

**Client PC details (please complete the following information)**
 - OS: Kubuntu 21.10, Manjaro KDE 21.2.5
 - Moonlight Version: 4.0.0
 - GPU: AMD Ryzen™ 7 4700U APU

**Server PC details (please complete the following information)**
 - OS: Windows 10 Pro  10.0.19043 Build 19043
 - GeForce Experience version: 3.25.1.27
 - Nvidia GPU driver: 512.59
 - Antivirus and firewall software: Windows Defender and Windows Firewall

**Moonlight Logs**
Recorded a new log file of the client streaming "Steam Big Picture" for a bit and then quitting and streaming "Hades" for a few minutes. 
[moonlight.log](https://github.com/moonlight-stream/moonlight-qt/files/8599614/moonlight.log)

**Additional context**
There's is no audio crackling when streaming games through Steam remote play or P****c. 

I also tried using Sunshine on the host PC instead of GeForce Experience, but it didn't help.

**Comment by cgutman** [comment]
Can you describe the kind of crackling? Was it like a constant static type of noise or just periodic crackles?

What audio output device are you using on your client?

Were you using the Snap, Flatpak, or AppImage version of Moonlight?

**Comment by selfdocumentingcode** [comment]
It's more like periodic crackling, I guess. It can also only be heard when there's sound coming from the host; if there's nothing playing, there's no crackling.

I attached a recording I made with my phone, although the phone doesn't capture it that well. (had to .zip it since github didn't allow me to attach .m4a)
[crackling.zip](https://github.com/moonlight-stream/moonlight-qt/files/8608351/crackling.zip)
 
I'm using hdmi output.

I tried both Snap and AppImage on Kubuntu and Flatpak on Manjaro. Experienced the exact same problem on all of them.

**Comment by cgutman** [comment]
You can try an alternate audio backend by setting an environment variable: `ML_AUDIO=libsoundio`.

You would run `flatpak run --env=ML_AUDIO=libsoundio com.moonlight_stream.Moonlight//stable` (Flatpak) or `ML_AUDIO=libsoundio moonlight` (Snap).

**Comment by selfdocumentingcode** [comment]
Alright, I will try it later when I get home and let you know how it went.

**Comment by selfdocumentingcode** [comment]
Alright, I will try it later when I get home and let you know how it went. 

LE: when launching moonlight with ML_AUDIO=libsoundio I would get this error message "Failed to open audio device. Audio will be unavailable during this session." when connecting to a stream. And there was indeed no sound.

Your suggestion gave me the idea to try using a different output so I connected a pair of headphones to the 3.5mm jack and there was no more crackling. 
This made me too look into hdmi audio crackling specifically and I found [this topic on askubuntu ](https://askubuntu.com/questions/405071/static-and-crackling-in-my-hdmi-audio)with a solution: to set tsched=0 somewhere in the PulseAudio config. 

A bit curious that I only encounter this problem when using Moonlight; I guess it's a rare edge case. 

Thanks for your help, btw. If you still want to look into this issue, I will gladly provide more info and/or test other configurations.
